lib/internal/Magento/Framework/Image/Adapter/ConfigInterface.php

```diff
@@ -20,4 +20,14 @@ public function getAdapterAlias();
      * @return array
      */
     public function getAdapters();
+
+    /**
+     * @return int
+     */
+    public function getMaxWidth();
```

Comment:
We can't add new methods to existing interfaces, it would be better to introduce a new interface and implement this methods in `Magento/Framework/Image/Adapter/Config.php`